EIPS/eip-5553.md

```diff
@@ -0,0 +1,260 @@
+
+```solidity  
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "./SongRegistration.sol";
+import "./CompositionRoyaltyToken.sol";
+import "./RecordingRoyaltyToken.sol";
+
+
+contract SimpleSongLedger is IERC721Receiver {
+    using Counters for Counters.Counter;
+    Counters.Counter private songIds;
+      function onERC721Received(address, address, uint256, bytes calldata) external pure returns (bytes4) {
+        return IERC721Receiver.onERC721Received.selector;
+    }
+
+    function mintSong(SongMintingParams memory _params) public {
+        CompositionRoyaltyToken comp = new CompositionRoyaltyToken(address(this),"SONGCOMP","COMP");
+        RecordingRoyaltyToken rec = new RecordingRoyaltyToken(address(this),"SONGREC","REC");
+        songIds.increment();
+
+        SongRegistration newSong = new SongRegistration(
+                                        songIds.current(),
+                                        address(this),
+                                        _params,
+                                        address(comp),
+                                        address(rec)
+                                    );
+    }
+}
+
+
+```
+
+## Rationale
+
+### Array of EIP-20 tokens presents a more robust royalty interest structure
+
+Current royalty implementations deal only with a single type of royalty payments: NFT sales. They also only allow a single type of royalty - i.e you cannot have different royalty interest owners receive royalty based on different licensing types. 
+In other words, currently, a royalty split works the same way no matter what type of purchase or license deal has happened, for all parties involved.
+
+With this proposal, we allow multiple **types** of royalty interests. A classic case is the music industry in which you have writing/composition royalties, and recording/master royalties. Different licensing types will pay different percentages to different parties based on context.
+
+In the case of a song cover, a license payment formula can be created so that that 
+a) Original work's writers get paid for using the lyrics or composition of the song
+b) recording artists of the original song do not get paid since their recording is not used
+c) recording artists of the new work will get paid
+d) there are no writing royalties for the creators of the cover.
+
+Moreover, we have single structure that connects to all types of royalty types, and allows finding them easily.
+Lastly, moving EIP-20 tokens around is much easier than managing an 0xsplits contract.
+
+### Separating the work from the sellable NFT enables scaling licensing types
+By separating the canonical version of the work from its various licensed uses (NFT purchase, streaming, usage of art etc..) we are allowing a new ecosystem of various license types and payment types for a single work, based on the same standard.
```

Comment:
```suggestion
By separating the canonical version of the work from its various licensed uses (NFT purchase, streaming, usage of art etc..) this EIP introduces a new ecosystem of various license types and payment types for a single work, based on the same standard.
```

Reply:
I like the direction, but it is not introducing a new ecosystem, but introducing a root for an ecosystem to evolve. Makes sense?